keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -128,16 +128,30 @@ class ImageService(
     appVersion: AppVersion,
     account: String,
     region: String
-  ): NamedImage? =
-    namedImageCache
-      .get(NamedImageCacheKey(appVersion, account, region))
-      .await()
+  ): NamedImage? {
+    val key = NamedImageCacheKey(appVersion, account, region)
+    return namedImageCache.get(key).await()
       // prime the cache if the image is also in other regions
       ?.also { image ->
         (image.amis.keys - region).forEach { otherRegion ->
-          namedImageCache.put(NamedImageCacheKey(appVersion, account, otherRegion), completedFuture(image))
+          namedImageCache.putIfMissingOrOlderThan(key.copy(region = otherRegion), image)
```

Comment:
Instead of _always_ adding the image to the cache for any other regions, only do so if it's not cached currently, or the current cached image is older than this one